@here/harp-mapview/lib/MapView.ts

```diff
@@ -2518,6 +2556,94 @@ export class MapView extends THREE.EventDispatcher {
         return this.m_elevationProvider;
     }
 
+    private extractAttitude() {
+        const camera = this.m_camera;
+        const projection = this.projection;
+
+        const cameraPos = cache.vector3[1];
+        const transform = cache.transform[0];
+        const tangentSpaceMatrix = cache.matrix4[1];
+        // 1. Build the matrix of the tangent space of the camera.
+        cameraPos.setFromMatrixPosition(camera.matrixWorld); // Ensure using world position.
+        projection.localTangentSpace(projection.unprojectPoint(cameraPos), transform);
+        tangentSpaceMatrix.makeBasis(transform.xAxis, transform.yAxis, transform.zAxis);
+
+        // 2. Change the basis of matrixWorld to the tangent space to get the new base axes.
+        cache.matrix4[0].getInverse(tangentSpaceMatrix).multiply(camera.matrixWorld);
+        transform.xAxis.setFromMatrixColumn(cache.matrix4[0], 0);
+        transform.yAxis.setFromMatrixColumn(cache.matrix4[0], 1);
+        transform.zAxis.setFromMatrixColumn(cache.matrix4[0], 2);
+
+        // 3. Deduce orientation from the base axes.
+        let yaw = 0;
+        let pitch = 0;
+        let roll = 0;
+
+        // Decompose rotation matrix into Z0 X Z1 Euler angles.
+        const epsilon = 1e-10;
+        const d = transform.zAxis.dot(cameraPos.set(0, 0, 1));
+        if (d < 1.0 - epsilon) {
+            if (d > -1.0 + epsilon) {
+                yaw = Math.atan2(transform.zAxis.x, -transform.zAxis.y);
+                pitch = Math.acos(transform.zAxis.z);
+                roll = Math.atan2(transform.xAxis.x, transform.yAxis.z);
+            } else {
+                // Looking bottom-up with space.z.z == -1.0
+                yaw = -Math.atan2(-transform.yAxis.x, transform.xAxis.x);
+                pitch = 180;
+                roll = 0;
+            }
+        } else {
+            // Looking top-down with space.z.z == 1.0
+            yaw = Math.atan2(-transform.yAxis.x, transform.xAxis.x);
+            pitch = 0.0;
+            roll = 0.0;
+        }
+
+        return {
+            yaw,
+            pitch,
+            roll
+        };
+    }
+
+    private lookAtImpl(params: LookAtParams): void {
+        const zoomLevel = THREE.MathUtils.clamp(
+            getOptionValue(params.zoomLevel, this.zoomLevel),
+            this.m_minZoomLevel,
+            this.m_maxZoomLevel
+        );
+        const tilt = Math.min(getOptionValue(params.tilt, this.tilt), PITCH_LIMIT_DEG);
```

Comment:
This will coerce custom pitch > 90 set by user manually to `PITCH_LIMIT_DEG` even if someone intended to change some other parameter like azimuth i. `lookAt({azimumuth: xxx})`.

Do we support camera with pitch > `PITCH_LIMIT_DEG` at all ? If not, then it's not documented.

Reply:
You could only get into `pitch > PITCH_LIMIT_DEG` situation if you are not using the `lookAt` API but modify the camera directly.
So in general we support it but we don't support mixing of direct camera modifications and `MapView.lookAt`. I think this is a valid restriction.

Reply:
Or let's put it differently: If you modify the camera directly there are no constraints, but when you use the lookAt API again the constraints for that API will be enforced.